api/signals.py

```diff
@@ -30,3 +42,61 @@ def track_application_actions(sender, instance, **kwargs):
             history.save()
         except ValueError:
             pass
+
+
+@receiver(post_save, sender=AssetIncidentReport)
+def create_notification_on_incident_report_submission(**kwargs):
+    """
+    Create a notification to admin users when a user creates an incident report
+    :param kwargs:
+    :return:
+    """
+    instance = kwargs["instance"]
+    # if a new notification has been  created
+    if kwargs.get("created", False) is True:
+        notification_body = "{} submitted a new incident report {}."
+        notification_title = "New Incident Report"
+        notification_targets = User.objects.filter(is_superuser=True)
+        notification_origin = instance.submitted_by
+        incident_url = "{}/{}".format(reverse("incidence-reports-list"), instance.id)
+
+        # create notifications to te different admins
+        for target in notification_targets:
```

Comment:
This is a good place to use [atomic commit](https://docs.djangoproject.com/en/2.2/topics/db/transactions/#django.db.transaction.atomic) to ensure this is done with a single call to the database in a transaction.

Reply:
Thanks for this feedback. This actually something I had not considered  👍